src/transformers/file_utils.py

```diff
@@ -142,6 +142,20 @@
 except (AttributeError, ImportError):
     _has_sklearn = False
 
+try:
+    # Test copied from tqdm.autonotebook: https://github.com/tqdm/tqdm/blob/master/tqdm/autonotebook.py
+    get_ipython = sys.modules["IPython"].get_ipython
+    if "IPKernelApp" not in get_ipython().config:
+        raise ImportError("console")
+    if "VSCODE_PID" in os.environ:
+        raise ImportError("vscode")
+
+    import IPython  # noqa: F401
+
+    _in_notebook = True
+except (ImportError, KeyError):
```

Comment:
Can we replace this with 
```
except (ImportError, KeyError, AttributeError):
```

? 

This can happen if `get_ipython()` has no `config`

@LysandreJik @sgugger

Reply:
Solved in https://github.com/huggingface/transformers/commit/2dba7d570257fc83ea5a1f82eb76939daa43fe35 by catching all errors, like it's done in the `tqdm.autonotebook` linked above.

Reply:
@sgugger feel free to update if you want to catch specific errors like @patrickvonplaten was proposing.

Reply:
I'm not in favor of having empty except clauses, so would prefer adding the `AttributeError`. I have a followup PR to fix the behavior when there is no evaluation, so will put that in it too.